core/Url.php

```diff
@@ -172,17 +172,40 @@ public static function getCurrentScriptName($removePathInfo = true)
             }
         }
 
-        if (!isset($url[0]) || $url[0] !== '/') {
-            $url = '/' . $url;
-        }
-
         // A hash part should actually be never send to the server, as browsers automatically remove them from the request
         // The same happens for tools like cUrl. While Apache won't answer requests that contain them, Nginx would handle them
         // and the hash part would be included in REQUEST_URI. Therefor we always remove any hash parts here.
         if (mb_strpos($url, '#')) {
             $url = mb_substr($url, 0, mb_strpos($url, '#'));
         }
 
+        // replace relative path references with absolute.
+        if (strlen($url) > 0) {
+            $urlSections = explode('/', $url);
+            $absoluteUrlComponents = [];
+
+            foreach ($urlSections as $section) {
+                if ($section === '.') {
+                    continue;
+                }
```

Comment:
We probably should skip an empty segment as well
```suggestion
                if (('.' === $section) || ('' === $section)) {
                    continue;
                }
```

Reply:
I added this check in, but I decided to remove it again because theres a few instances where picking up on the empty strings at either end of a path `/test/path/`, ensures the correct extra slashes get added. I've instead added a step where consecutive slashes are removed `'//' => '/'`.